lib/cosmos/tools/tlm_grapher/data_objects/data_object.rb

```diff
@@ -101,17 +101,33 @@ def process_packet(packet, count)
       raise "process_packet must be defined by class #{self.class}"
     end
 
+    def handle_process_exception(error, telemetry_point)
+      raise error if error.class == NoMemoryError
+      reset()
+      @plot.redraw_needed = true
+      @error = error
+      if error.is_a? TypeError
+        @error = FatalError.new("Telemetry point #{telemetry_point} could not be displayed. This is most likely due to this telemetry point being a String which can't be represented on the graph. Remove the point from the list of telemetry items to cause this exception to go away.\n\n#{error}")
+      else
+        @exceptions_reported ||= []
+        unless @exceptions_reported.include?(error.message)
+          @exceptions_reported << error.message
+          Cosmos.write_exception_file(error)
```

Comment:
I ensure we write at least one exception file whether the user clicks on the error icon or not.